core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -84,6 +122,8 @@ case object MappedPoolMemory extends MBeanExecutorMetricType(
   "java.nio:type=BufferPool,name=mapped")
 
 private[spark] object ExecutorMetricType {
+  final val pTreeInfo = new ProcfsBasedSystems
```

Comment:
Can ProcfsBasedSystems just be an `object` in and of itself?

Reply:
this is a weird place to keep this, unless there is some really good reason for it.  I think it should go inside `ProcessTreeMetrics`.

also I'm not sure what the problem was with making it an object.  Seems to work for me.  its a bit different now as there are arguments to the constructor for testing -- but you could still have an object which extends the class

```scala
private[spark] object ProcfsBasedSystems extends ProcfsBasedSystems("/proc/")
```

though doesn't really seem to have much value.

Reply:
What are the benefits of the companion object vs this current approach? I can revert to the companion object model and do testing again to see what was the problem before, but just wanted to understand the benefits of it before investing time.

Reply:
Normally having an object helps make it clear that there is a singleton; its easier to share properly and easier to figure out how to get a handle on it.  Given that we'll have a class anyway, I don't think there is a ton of value in having there be a companion object.

I do still think the instance you create here should go somewhere else.